airbyte-webapp/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,234 @@
+  }
+
+  let numeratorRecords = -1;
+  let denominatorRecords = -1;
+  let totalPercentRecords = -1;
+  let numeratorBytes = -1;
+  let denominatorBytes = -1;
+  let elapsedTimeMS = -1;
+  let timeRemaining = -1;
+  let timeRemainingString = "";
+  const unEstimatedStreams: string[] = [];
+  let displayProgressBar = true;
+  let latestAttempt: AttemptRead | undefined;
+
+  const jobStatus = getJobStatus(job);
+
+  // colors from `_colors.scss` TODO: Use the SCSS variables maybe?
+  let color = "white";
+  switch (jobStatus) {
+    case "pending":
+      color = "#cbc8ff";
+      break;
+    case "running":
+      color = "#cbc8ff";
+      break;
+    case "incomplete":
+      color = "#fdf8e1";
+      break;
+    case "failed":
+      color = "#e64228";
+      return null;
+    case "succeeded":
+      color = "#67dae1";
+      return null;
+    case "cancelled":
+      return null;
+  }
+
+  if (isJobsWithJobs(job) && job.attempts) {
+    latestAttempt = job.attempts[job.attempts?.length - 1];
+    let countTotalsFromStreams = true;
+    if (
+      latestAttempt.totalStats?.recordsEmitted &&
+      latestAttempt.totalStats?.estimatedRecords &&
+      latestAttempt.totalStats?.bytesEmitted &&
+      latestAttempt.totalStats?.estimatedBytes
+    ) {
+      countTotalsFromStreams = false;
+      numeratorRecords = latestAttempt.totalStats.recordsEmitted;
+      denominatorRecords = latestAttempt.totalStats.estimatedRecords;
+      numeratorBytes = latestAttempt.totalStats.bytesEmitted;
+      denominatorBytes = latestAttempt.totalStats.estimatedBytes;
+    }
+
+    if (latestAttempt && !latestAttempt.totalStats && latestAttempt.streamStats) {
+      for (const stream of latestAttempt.streamStats) {
+        if (!stream.stats.recordsEmitted) {
+          unEstimatedStreams.push(`${stream.streamName}`);
+        }
+        if (countTotalsFromStreams) {
+          numeratorRecords += stream.stats.recordsEmitted ?? 0;
+          denominatorRecords += stream.stats.estimatedRecords ?? 0;
+          numeratorBytes += stream.stats.bytesEmitted ?? 0;
+          denominatorBytes += stream.stats.estimatedBytes ?? 0;
+        }
+      }
+    } else if (!latestAttempt.streamStats) {
+      displayProgressBar = false;
+    }
+  }
+
+  totalPercentRecords = denominatorRecords > 0 ? Math.floor((numeratorRecords * 100) / denominatorRecords) : 0;
+
+  // chose to estimate time remaining based on records rather than bytes
+  if (latestAttempt && latestAttempt.status === Status.RUNNING && displayProgressBar) {
+    elapsedTimeMS = new Date().getTime() - latestAttempt.createdAt * 1000;
+    timeRemaining = Math.floor(elapsedTimeMS / totalPercentRecords) * (100 - totalPercentRecords); // in ms
+    const minutesRemaining = Math.ceil(timeRemaining / 1000 / 60);
+    const hoursRemaining = Math.ceil(minutesRemaining / 60);
+    if (minutesRemaining <= 60) {
+      timeRemainingString = `${minutesRemaining} ${formatMessage({ id: "estimate.minutesRemaining" })}`;
```

Comment:
General note how i18n works (this applies to most strings in this PR, so I will only attach it here in one example place):

If you need to concatenate a string, with a value, the value needs to be a placeholder in the actual string and passed in as a value. Concatenating outside of the i18n framework would assume that all languages need the number at the beginning of the phrase. So this message should be:

```ts
timeRemainingString = formatMessage({ id: "estimate.minutesRemaining" }, { minutes: minutesRemaining });
```

and the message would be (we also should use correct pluralization rules inside to end up with proper grammar):

```
{plural, minutes, one {# minute} other {# minutes}} remaining
```